identity-service/src/relay/txRelay.js

```diff
@@ -242,6 +216,51 @@ const sendTransactionInternal = async (req, web3, txProps, reqBodySHA) => {
   return txReceipt
 }
 
+const filterReplicaSetUpdates = (decodedABI, senderAddress) => {
+  // Rate limit replica set reconfiguration transactions
+  // A reconfiguration (as opposed to a first time selection) will have an
+  // _oldPrimaryId value of "0"
+
+  let isReplicaSetTransaction = false
+  let isFirstReplicaSetConfig = false
+
+  if (decodedABI.name === 'updateReplicaSet') {
+    // TODO remove legacy replica set updates
+    isReplicaSetTransaction = decodedABI.name === 'updateReplicaSet'
+    if (isReplicaSetTransaction) {
+      isFirstReplicaSetConfig = decodedABI.params.find(
+        (param) => param.name === '_oldPrimaryId' && param.value === '0'
+      )
+    }
+  } else if (decodedABI.name === 'EntityManager') {
```

Comment:
i'm confused how on L227 `decodedABI.name` is the function name while here it is the contract name?

Reply:
couldn't you check `contractName` here instead?

Reply:
ah yea this is wrong